osquery/events/linux/bpf/processcontextfactory.cpp

```diff
@@ -240,12 +240,12 @@ bool ProcessContextFactory::getParentPidFromStatFile(IFilesystem& fs,
     return false;
   }
 
-  auto parent_pid_r_it = std::find(buffer.rbegin(), buffer.rend(), ')');
-  if (parent_pid_r_it == buffer.rend()) {
+  auto parent_pid_it = std::find(buffer.begin(), buffer.end(), ')');
+  if (parent_pid_it == buffer.end()) {
```

Comment:
Careful that a process can contain a `)` in its name, which would create an issue here; this is why the code tries to find the last occurrence by searching from the end.